omegaconf/listconfig.py

```diff
@@ -241,7 +241,32 @@ def _set_at_index(self, index: Union[int, slice], value: Any) -> None:
 
     def __setitem__(self, index: Union[int, slice], value: Any) -> None:
         try:
-            self._set_at_index(index, value)
+            if isinstance(index, slice):
+                _ = iter(value)  # check iterable
+                indexes = range(*index.indices(len(self)))
+                if index.step is not None:
```

Comment:
```suggestion
                if index.step not in (None, 1):
```
This makes `ListConfig.__setitem__` behave like `list.__setitem__` in the edge case where `step == 1`.